### PR TITLE
Python 3 support

### PR DESCRIPTION
**Pull request by flit**
Numerous changes in order to support running pyOCD in Python 3.

The biggest areas of change are:

1. The most involved changes are in the gdb server, where all `str` objects were changed to `bytes`. 
2. The other most noticeable change is that all module imports within pyOCD must now be relative.
3. Replaced all uses of `has_key()` with the `in` or `not in` operators.

All tests pass both on Python 2.7.15 and 3.6.5. Travis CI config is updated to test on 2.7, 3.6.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -18,14 +18,23 @@
...
+    'colorama',
+    ]
 if sys.platform.startswith('linux'):
     install_requires.extend([
         'pyusb>=1.0.0b2',
     ])
 elif sys.platform.startswith('win'):
     install_requires.extend([
         'pywinusb>=0.4.0',
+        'hidapi',
```
Comment (c1728p9) [inline setup.py]: Why is `hidapi` needed on windows? Shouldn't `pywinusb` cover windows?

File: test/automated_test.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ def main():
     test_time = (stop - start)
 
     print_summary(test_list, result_list, test_time)
-    with open(summary_file, "wb") as output_file:
+    with open(summary_file, "w") as output_file:
```
Comment (c1728p9) [inline test/automated_test.py]: was this causing problems or just changed since it isn't a binary file?

File: test/gdb_script.py [hunk 3]
```diff
@@ -69,16 +77,16 @@
     "help",
     "help reset",
     "help halt",
-    "help resume",
     "init",
     "reset",
     "reset halt",
-    "reset halt resume",
```
Comment (c1728p9) [inline test/gdb_script.py]: why were these removed?

File: test/gdb_script.py [hunk 4]
```diff
@@ -96,11 +104,16 @@
 }
 
 
+def gdb_execute(cmd):
+    print("Executing command:", cmd)
```
Comment (c1728p9) [inline test/gdb_script.py]: Was this intentionally left in or is it just for debugging?

File: setup.py (hunk 1 above)
Comment (flit) [reply setup.py]: It was added because there are issues with pywinusb on some boards.
Will remove and deal with separately.

File: test/automated_test.py (hunk 2 above)
Comment (flit) [reply test/automated_test.py]: In Python 3, files open in binary mode only accept bytes-like objects in their `write()` method, and return `bytes` from `read()`.

File: test/gdb_script.py (hunk 3 above)
Comment (flit) [reply test/gdb_script.py]: The "resume" monitor commands were disabled back in 2015.

File: test/gdb_script.py (hunk 4 above)
Comment (flit) [reply test/gdb_script.py]: Intentional. It's helpful to have the gdb commands listed in the `output.txt` file.
